src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlFileGenerator.cs

```diff
@@ -4904,63 +4908,80 @@ private string BuildBindingOption(XamlMemberDefinition m, INamedTypeSymbol? prop
 					return "null";
 				}
 
-				if (m.Member.Name == "TargetNullValue" && FindType(bindingType.Type) is INamedTypeSymbol namedTypeSymbol &&
-					m.Objects.SingleOrDefault()?.Members?.SingleOrDefault()?.Value is { } value)
-				{
-					return BuildLiteralValue(namedTypeSymbol, value.ToString());
-				}
-
 				if (IsCustomMarkupExtensionType(bindingType.Type))
 				{
 					return GetCustomMarkupExtensionValue(m);
 				}
 
-				// If type specified in the binding was not found, log and return an error message
-				if (!string.IsNullOrEmpty(bindingType.Type.Name))
+				if (memberName == "Converter" &&
+					m.Objects.SingleOrDefault() is XamlObjectDefinition { } converterObjectDefinition)
 				{
-					var message = $"#Error // {bindingType.Type.Name} could not be found.";
+					var fullTypeName = converterObjectDefinition.Type.Name;
+					var knownType = FindType(converterObjectDefinition.Type);
+					if (knownType == null && converterObjectDefinition.Type.PreferredXamlNamespace.StartsWith("using:"))
+					{
+						fullTypeName = converterObjectDefinition.Type.PreferredXamlNamespace.TrimStart("using:") + "." + converterObjectDefinition.Type.Name;
+					}
+					if (knownType != null)
+					{
+						// Override the using with the type that was found in the list of loaded assemblies
+						fullTypeName = knownType.ToDisplayString();
+					}
 
-					return message;
+					return $"new {GetGlobalizedTypeName(fullTypeName)}()";
 				}
 
-				return "#Error";
-			}
-			else
-			{
-
-				var value = BuildLiteralValue(m, GetPropertyType("Binding", memberName));
-
-				if (memberName == "Path")
+				if (FindType(bindingType.Type) is INamedTypeSymbol namedTypeSymbol &&
+					m.Objects.SingleOrDefault()?.Members?.SingleOrDefault() is XamlMemberDefinition { } innerMember)
 				{
-					value = RewriteAttachedPropertyPath(value);
+					m = innerMember;
 				}
-				else if (memberName == "ElementName")
+				else
 				{
-					if (m.Value == null)
+					// If type specified in the binding was not found, log and return an error message
+					if (!string.IsNullOrEmpty(bindingType.Type.Name))
 					{
-						throw new InvalidOperationException($"The property ElementName cannot be empty");
+						var message = $"null\r\n#error Invalid binding: {bindingType.Type.Name} could not be found.\r\n";
+
+						return message;
 					}
 
-					// Skip the literal value, use the elementNameSubject instead
-					string elementName = m.Value.ToString() ?? "";
-					value = "_" + elementName + "Subject";
-					// Track referenced ElementNames
-					CurrentScope.ReferencedElementNames.Add(elementName);
+					return "null\r\n#error Invalid binding\r\n";
```

Comment:
Let's add some additional information here, like the location of the binding expression from the original `m` XamlMember definition.